### ISSUE TITLE
Accept .im4p kernel cache, device tree, static trust cache files

### ISSUE THREAD

**Issue by qmfrederik** [TrungNguyen1909/qemu-t8030#7]
We're currently required to extract the kernel image, device tree and static trust cache from the .img4 files.

It may be useful if qemu itself can take care of the extraction - it would make the process easier.
Qemu already links with OpenSSL and liblzfse, so it already has the infrastructure required to manipulate im4p files in place.

Would you be open to a PR which adds such a feature?

**Comment by TrungNguyen1909** [comment]
Resolved in PR #9

**Comment by TrungNguyen1909** [comment]
Trust cache im4p has 2 types of payload_type:

`rtsc` for ramdisk's trustcache
`trst` for rootdisk's trustcache

```
qemu-system-aarch64: Could parse ASN.1 data in file 'Firmware/038-44087-125.dmg.trustcache' because it is not a 'trst' object, found 'rtscIM4P' object.
```

**Comment by TrungNguyen1909** [comment]
I suggest leaving `payload_type` check to caller of `extract_im4p_payload`

**Comment by TrungNguyen1909** [comment]
Addressed in #a596748
